dotnet/src/Microsoft.AutoGen/Agents/InMemory/InMemoryAgentWorkerRuntime.cs

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// InMemoryAgentWorkerRuntime.cs
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using Microsoft.AutoGen.Abstractions;
+using Microsoft.AutoGen.Runtime;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AutoGen.Agents;
+
+public class InMemoryAgentWorkerRuntime : IAgentWorkerRuntime, IAgentWorkerRegistryGrain, IWorkerGateway, IAgentContext
```

Comment:
Is it a good time to add tests for this class now

Reply:
so much still evolving... let's land it and then add tests.